pkg/providers/aws/cluster_network_provider.go

```diff
@@ -1110,32 +1110,62 @@ func (n *NetworkProvider) reconcileRDSVpcConfiguration(ctx context.Context, priv
 		return errorUtil.Wrap(err, "error building subnet group name")
 	}
 
-	foundSubnet, err := getRDSSubnetGroup(n.RdsApi, subnetGroupName)
-	if err != nil {
-		return errorUtil.Wrap(err, "failed getting rds subnet group")
-	}
-	if foundSubnet != nil {
-		logger.Infof("subnet group %s found", *foundSubnet.DBSubnetGroupName)
-		// todo ensure subnets and tags are correct
-		return nil
+	// build array list of all vpc private subnets
+	var subnetIDs []*string
+	for _, subnet := range privateVPCSubnets {
+		subnetIDs = append(subnetIDs, subnet.SubnetId)
 	}
 
 	// in the case of no private subnets being found, we return a less verbose error message compared to obscure aws error message
 	if len(privateVPCSubnets) == 0 {
 		return errorUtil.New("no private subnets found, can not create subnet group for rds")
 	}
 
-	// build array list of all vpc private subnets
-	var subnetIDs []*string
-	for _, subnet := range privateVPCSubnets {
-		subnetIDs = append(subnetIDs, subnet.SubnetId)
-	}
-
 	clusterID, err := resources.GetClusterID(ctx, n.Client)
 	if err != nil {
 		return errorUtil.Wrap(err, "error getting clusterID")
 	}
 
+	foundSubnet, err := getRDSSubnetGroup(n.RdsApi, subnetGroupName)
+	if err != nil {
+		return errorUtil.Wrap(err, "failed getting rds subnet group")
+	}
+	if foundSubnet != nil {
+		logger.Infof("subnet group %s found", *foundSubnet.DBSubnetGroupName)
+		if foundSubnet.DBSubnetGroupDescription != aws.String(defaultSubnetGroupDesc) {
+			_, err := n.RdsApi.ModifyDBSubnetGroup(&rds.ModifyDBSubnetGroupInput{
+				DBSubnetGroupDescription: aws.String(defaultSubnetGroupDesc),
+				DBSubnetGroupName: aws.String(*foundSubnet.DBSubnetGroupName),
+				SubnetIds: subnetIDs,
+			})
+			if err != nil {
+				return errorUtil.Wrap(err, "error updating db subnet group description")
+			}
+		}
+
+		tags, err := n.RdsApi.ListTagsForResource(&rds.ListTagsForResourceInput{
+			ResourceName: aws.String(*foundSubnet.DBSubnetGroupArn),
+		})
+		if err != nil {
+			return errorUtil.Wrap(err, "error getting subnet group tags")
+		}
+		for _, tag := range tags.TagList {
+			if tag.Key == aws.String("cluster") && tag.Value != aws.String(clusterID) {
```

Comment:
I think this `cluster` key is incorrect, we should be using the the `croOwnerTag`, can we update it here and on creation of the subnet group.